### PR TITLE
allow_to_customize_address_for_jerry_mem_heap

### PR DESCRIPTION
**Pull request by tczene**
JerryScript-DCO-1.0-Signed-off-by: Tamás Czene tczene.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by tczene** [comment]
Added new define to set memory address of Jerry heap.
More details: [#306](https://github.com/Samsung/iotjs/issues/306)

File: jerry-core/mem/mem-heap.cpp [hunk 1]
```diff
@@ -188,7 +188,7 @@ typedef struct
 /**
  * Heap
  */
-mem_heap_t mem_heap;
+mem_heap_t mem_heap JERRY_HEAP_SECTION_ATTR;
```
Comment (ruben-ayrapetyan) [inline jerry-core/mem/mem-heap.cpp]: @tczene, could you, please, describe how the option would be used?

Comment (tczene) [inline jerry-core/mem/mem-heap.cpp]: The ```JERRY_HEAP_SECTION_ATTR``` is set to ```__attribute__((section(.jerry_mem)))``` by IoT.js build script. With ```[variable] JERRY_HEAP_SECTION_ATTR``` expression we can set the ```[variable]``` to use section ```.jerry_mem```. The .jerry_mem is defined in nuttx linker script.

With this we can force the jerry-heap to take place in sram.

Comment (ruben-ayrapetyan) [inline jerry-core/mem/mem-heap.cpp]: @tczene, thank you for the description.

Maybe, we could set the option not to `__attribute__((section(.jerry_mem)))`, but to `.jerry_mem`, so simplifying scripts and code.
In the way, we could provide `.bss` as default value for the macro.

What do you think about this?

**Comment by egavrin** [comment]
Please fix precommit

File: jerry-core/config.h [hunk 2]
```diff
@@ -48,6 +48,13 @@
 #endif /* !CONFIG_MEM_HEAP_AREA_SIZE */
 
 /**
+ * Address of heap
+ */
+#ifndef JERRY_HEAP_SECTION_ATTR
```
Comment (ruben-ayrapetyan) [inline jerry-core/config.h]: Comment is incorrect.

**Comment by egavrin** [comment]
Good to me

File: jerry-core/mem/mem-heap.cpp [hunk 3]
```diff
@@ -188,7 +188,7 @@ typedef struct
 /**
  * Heap
  */
-mem_heap_t mem_heap;
+mem_heap_t mem_heap __attribute__ ((section (JERRY_HEAP_SECTION_ATTR)));
```
Comment (zherczeg) [inline jerry-core/mem/mem-heap.cpp]: I would put the ifdef here. It is true, that this variable will likely go to .bss, but I would not force this.
```
#ifndef JERRY_HEAP_SECTION_ATTR
mem_heap_t mem_heap;
#else
mem_heap_t mem_heap __attribute__ ((section (JERRY_HEAP_SECTION_ATTR)));
#endif
```

**Comment by egavrin** [comment]
@galpeter @zherczeg please review this pr

**Comment by zherczeg** [comment]
LGTM

**Comment by seanshpark** [comment]
ping

**Comment by egavrin** [comment]
Merged f8f7eb42511947b41f475312afc5fd057a902064

**Pull request closed without merging**